routing/doc.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2015 Zalando SE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+Package routing implements matching of http requests to a continuously
+updatable set of skipper routes.
+
+
+Request Evaluation
+
+1. The path in the http request is used to find one or more matching
+route definitions in a lookup tree.
+
+2. The rest of the request attributes is matched against the non-path
+conditions of the routes found in the lookup tree, from the most to the
+least strict one. The result is the first route whose every condition is
+met.
+
+(The regular expression conditions for the path, 'PathRegexp', are
+applied only in step 2.)
+
```

Comment:
Good place to add the reference to the regex dialect and the link to the Go specs